### PR TITLE
Add precommit hooks

### LINKED ISSUE #6: Add pre-commit hooks for local linting
Something that will be nice to have, and helps a _lot_ in development, is the user of [`pre-commit`](https://pre-commit.com/) and `pre-commit` hooks to perform local linting and checks while you're working on a PR. At the moment, there isn't much code in the project so this might seem like overkill, but it helps considerably once projects scale up.

It would be nice to have:

- A `dev-requirements.txt` file that has tooling that will be helpful to install if doing dev work (so to start off with just) (note here it doesn't matter as much the pinning to versions)

```
$ cat dev-requirements.txt
pre-commit 
```
- A `.pre-commit-config.yaml` to contain the pre-commit hooks. c.f. https://scikit-hep.org/developer/style#pre-commit and [`pyhf`'s `.pre-commit-config.yaml`](https://github.com/scikit-hep/pyhf/blob/5dc122ada38ceb010b2ea60b43cc837b84beb5e8/.pre-commit-config.yaml) for an example

- Setup https://pre-commit.ci/ for the repo so that it the hooks will get run in CI on PRs as well.

### PR DESCRIPTION
**Pull request by mihirkatare**
Close #6

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+ci:
+  autoupdate_commit_msg: "chore: [pre-commit.ci] pre-commit autoupdate"
+  autoupdate_schedule: "monthly"
```
Comment (matthewfeickert) [inline .pre-commit-config.yaml:3]:
```suggestion
  autoupdate_schedule: monthly
```

I think these might need to be unquoted (given that I had this quoted up till today and noticed that `pyhf`'s hadn't been updated in over a month). Not 100% sure on this being required, but I know it works if it is unquoted.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
```
Comment (matthewfeickert) [inline .pre-commit-config.yaml:7]:
```suggestion
    rev: v4.0.1
```

`v4.0.1` is out now. You can get `pre-commit` to update for you by doing (locally)

```console
$ pre-commit autoupdate
```

which is really nice. :)

**Comment by matthewfeickert** [comment]
I like badges (perhaps a bit too much) but it would be nice to have a `pre-commit.ci` badge under the NSF funding one. 

```markdown
[![pre-commit.ci status](https://results.pre-commit.ci/badge/github/mihirkatare/DeepMEM/main.svg)](https://results.pre-commit.ci/latest/github/mihirkatare/DeepMEM/main)
```
[pre-commit.ci status](https://results.pre-commit.ci/latest/github/mihirkatare/DeepMEM/main)

File: .pre-commit-config.yaml (hunk 2 above)
Comment (mihirkatare) [reply .pre-commit-config.yaml:7]: Thanks for the tip, this is actually really convenient.

**Review by matthewfeickert — APPROVED** [review]
LGTM :shipit:

**Pull request merged**